ara/core/result.h

```diff
@@ -0,0 +1,382 @@
+#ifndef ARA_CORE_RESULT_H_
+#define ARA_CORE_RESULT_H_
+
+#include <type_traits>
+#include <boost/variant.hpp>
+#include "error_code.h"
+
+namespace ara {
+namespace core {
+
+template <typename T, typename E = ErrorCode>
+class Result final
+{
+    boost::variant<T, E> _data;
+
+public:
+    using value_type = T;
+    using error_type = E;
+
+    Result(T const &t): _data(t) {};
+    Result(T &&t): _data(std::move(t)) {};
+    explicit Result(E const &e): _data(e) {};
+    explicit Result(E &&e): _data(std::move(e)) {};
+    Result(Result const &other): _data(other._data) {};
+    Result(Result &&other) noexcept(std::is_nothrow_move_constructible<T>::value &&
+                                    std::is_nothrow_move_constructible<E>::value)
+        : _data(std::move(other._data)) {};
+    ~Result();
+
+    static Result FromValue(T const &t)
+    {
+        return Result(t);
+    }
+
+    static Result FromValue(T &&t)
+    {
+        return Result(std::move(t));
+    }
+
+    template <typename... Args>
+    static Result FromValue(Args &&... args)
+    {
+        return Result(T{std::forward<Args>(args)...});
+    }
+
+    static Result FromError(E const &e)
+    {
+        return Result(e);
+    }
+
+    static Result FromError(E &&e)
+    {
+        return Result(std::move(e));
+    }
+
+    template <typename... Args>
+    static Result FromError(Args &&... args)
+    {
+        return Result(E{std::forward<Args>(args)...});
+    }
+
+    Result& operator= (Result const &other)
```

Comment:
redundant whitespace between `operator=` and `(...)`

Reply:
done for all